### PR TITLE
[opt](log) refactor the log dir config

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

Refactor the config for log dir of FE and BE

TLDR:
- Use env variable `LOG_DIR` to set root log dir
- Remove `sys_log_dir` for FE and BE

Details:

1. FE

    1. The root log dir is set by env variable `LOG_DIR` in `fe.conf`
    2. The default value of `audit_log_dir` is same as `${LOG_DIR}/`
    3. The default value of `spark_launcher_log_dir` is `${LOG_DIR}/spark_launcher_log`
    4. The default value of `nereids_trace_log_dir` is `${LOG_DIR}/nereids_trace_log`
    5. The origin `sys_log_dir` is deprecated, and default value is `""`.
        But for compatibility, if user already set `sys_log_dir` before, Doris will still use it as root log dir.

2. BE

     1. The root log dir is set by env variable `LOG_DIR` in `be.conf`
     2. Remove `pipeline_tracing_log_dir`, use `${LOG_DIR}` directly.
     3. The origin `sys_log_dir` is deprecated, and default value is `""`.
         But for compatibility, if user already set `sys_log_dir` before, Doris will still use it as root log dir.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: be/src/pipeline/pipeline_tracing.cpp [hunk 1]
```diff
@@ -95,10 +95,11 @@ void PipelineTracerContext::_dump(TUniqueId query_id) {
         return;
     }
 
+    std::filesystem::path log_dir = getenv("LOG_DIR");
     //TODO: when dump, now could append records but can't add new query. try use better grained locks.
     std::unique_lock<std::mutex> l(_data_lock); // can't rehash
     if (_dump_type == RecordType::PerQuery) {
-        auto path = _dir / fmt::format("query{}", to_string(query_id));
+        auto path = log_dir / fmt::format("query{}", to_string(query_id));
```
Comment (Gabriel39) [inline be/src/pipeline/pipeline_tracing.cpp:102]: Lost `tracing` dir

Comment (morningman) [reply be/src/pipeline/pipeline_tracing.cpp:102]: done

**Comment by morningman** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

File: be/src/pipeline/pipeline_tracing.h [hunk 2]
```diff
@@ -52,6 +52,7 @@ struct ScheduleRecord {
 using OneQueryTraces = moodycamel::ConcurrentQueue<ScheduleRecord>;
 
 // belongs to exec_env, for all query, if enable
+// curl http://{host}:{web_server_port}/api/running_pipeline_tasks
```
Comment (zclllyybb) [inline be/src/pipeline/pipeline_tracing.h:55]: `PipelineTracerContext` has nothing to do with running_pipeline_tasks. It's a experimental function WIP

**Review by zddr — APPROVED** [review]

**Review by AshinGau — APPROVED** [review]
